### ISSUE TITLE
Warmup training for new output layer

### ISSUE THREAD

**Issue by aanchan** [SlangLab-NU/PSST_Training#6]
WWW
As a machine learning researcher training new models, our training code needs to be updated in order to enable training so that when a new output layer is added, the rest of the layers are frozen for the first few thousand epochs. After this so-called linear warmup all but the encoder layers can continue to be trained. This comes included in fairseq but not in HuggingFace training tools.

The following link should help.
Learning rate scheduler: HF should use a linear scheduler, warm-up is important
Trouble with implementing layer freezing during training : https://discuss.huggingface.co/t/gradual-layer-freezing/3381

AC
- A PR on the training code with documentation of a successful training run.
